streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -1232,47 +1231,51 @@ public void shouldRespectPunctuateCancellationSystemTime() {
         processorSystemTime.mockProcessor.checkAndClearPunctuateResult(PunctuationType.WALL_CLOCK_TIME, now + 10);
     }
 
-    @Test
-    public void shouldRespectCommitNeeded() {
-        when(stateManager.taskId()).thenReturn(taskId);
-        when(stateManager.taskType()).thenReturn(TaskType.ACTIVE);
-        task = createSingleSourceStateless(createConfig(AT_LEAST_ONCE, "0"), StreamsConfig.METRICS_LATEST);
-        task.initializeIfNeeded();
-        task.completeRestoration(noOpResetter -> { });
-
-        assertFalse(task.commitNeeded());
-
-        task.addRecords(partition1, singletonList(getConsumerRecordWithOffsetAsTimestamp(partition1, 0)));
-        assertTrue(task.process(0L));
-        assertTrue(task.commitNeeded());
-
-        task.prepareCommit();
-        assertTrue(task.commitNeeded());
-
-        task.postCommit(true);
-        assertFalse(task.commitNeeded());
-
-        assertTrue(task.canPunctuateStreamTime());
-        assertTrue(task.maybePunctuateStreamTime());
-        assertTrue(task.commitNeeded());
-
-        task.prepareCommit();
-        assertTrue(task.commitNeeded());
-
-        task.postCommit(true);
-        assertFalse(task.commitNeeded());
-
-        time.sleep(10);
-        assertTrue(task.canPunctuateSystemTime());
-        assertTrue(task.maybePunctuateSystemTime());
-        assertTrue(task.commitNeeded());
-
-        task.prepareCommit();
-        assertTrue(task.commitNeeded());
-
-        task.postCommit(true);
-        assertFalse(task.commitNeeded());
-    }
+//    @Test
```

Comment:
Reminder to not approve this as long as we have commented out test cases.

Reply:
Oh, it was mistakenly commented out!